src/sbd-inquisitor.c

```diff
@@ -784,52 +784,49 @@ parse_device_line(const char *line)
     int last = 0;
     int max = 0;
     int found = 0;
+    int skip_space = 1;
+    int space_run = 0;
 
-    if(line) {
-        max = strlen(line);
+    if (!line) {
+        return 0;
     }
 
-    if (max <= 0) {
-        return found;
-    }
+    max = strlen(line);
 
     cl_log(LOG_DEBUG, "Processing %d bytes: [%s]", max, line);
-    /* Skip initial whitespace */
-    for (lpc = 0; lpc <= max && isspace(line[lpc]); lpc++) {
-        last = lpc + 1;
-    }
 
-    /* Now the actual content */
     for (lpc = 0; lpc <= max; lpc++) {
-        int a_space = isspace(line[lpc]);
-
-        if (a_space && lpc < max && isspace(line[lpc + 1])) {
-            /* fast-forward to the end of the spaces */
-
-        } else if (a_space || line[lpc] == ';' || line[lpc] == 0) {
-            int rc = 1;
-            char *entry = NULL;
+        if (isspace(line[lpc])) {
+            if (skip_space) {
+                last = lpc + 1;
+            } else {
+                space_run++;
+            }
+            continue;
+        }
+        skip_space = 0;
+        if (line[lpc] == ';' || line[lpc] == 0) {
+            int rc = 0;
+            char *entry = calloc(1, 1 + lpc - last);
 
-            if (lpc > last) {
-                entry = calloc(1, 1 + lpc - last);
-                if (!entry) {
-                    fprintf(stderr, "heap allocation failed parsing device-line.\n");
-                    exit(1);
-                }
+            if (entry) {
```

Comment:
I tend to think we'd want a fatal exit if memory allocation fails, otherwise we'll just log a warning and possibly use a partial set of devices

Reply:
There is already a bail-out-path for if it doesn't find at least one valid entry. So it is probably easiest to use that for oom as well. Question is - actually for the same reason not to run with a partial config - if we really should be content with detecting one valid entry or if certain parsing-issues shouldn't lead to a bail-out on top.